lib/vmdb/settings/database_source.rb

```diff
@@ -76,8 +76,9 @@ def resource_queryable?
       end
 
       def database_connectivity?
-        conn = ActiveRecord::Base.connection rescue nil
-        conn && ActiveRecord::Base.connected?
+        ActiveRecord::Base.connection && ActiveRecord::Base.connected?
+      rescue ActiveRecord::NoDatabaseError
+        false
```

Comment:
I think the other case we want to do is this (as I was working on the same thing 😉 ):

```ruby
def database_connectivity?
  ActiveRecord::Base.connection && ActiveRecord::Base.connected?
rescue ActiveRecord::NoDatabaseError
  false
rescue => e
  return false if e.message =~ /^Could not load database configuration\./
end
```

From here:

https://github.com/rails/rails/blob/fe76a95b0d252a2d7c25e69498b720c96b243ea2/railties/lib/rails/application/configuration.rb#L241

(and there is no specified error... so you have to match on message...)

Reply:
Yeah, I'll add that.  Only problem is how to verify it?  Did you run a specific rake task? (rake environment blows up otherse)

Reply:
If I rm config/database.yml, I get a different error - 

https://github.com/rails/rails/blob/fe76a95b0d252a2d7c25e69498b720c96b243ea2/railties/lib/rails/application/configuration.rb#L250

Reply:
@Fryguy I did a `mv config/database.yml config/database.yml.bak` and then tested it by just doing:

```console
$ bin/rails r "foo = 1"
```

That said, the regexp could probably just change to:

```ruby
/^(?:Could |Can)not load database configuration(?:\.|:)/
```

Which should cover both errors.

Reply:
@Fryguy Oh crap... my code up there, I forgot to add another `raise` right after the `return false`, so it would re-raise if it doesn't catch the specific error.

My bad, it is what I had locally, but didn't put it here.

Reply:
But I can't even get to that point...it blows up before this code is even hit.  In fact, the code doesn't even get to initializing the settings as it blows up way before this line, which is where the Settings are loaded first.

https://github.com/ManageIQ/manageiq/blob/0113603976752db6ca36aa925f6b8dd60428376c/config/application.rb#L161

Reply:
@NickLaMuro I actually added that raise locallly cause I noticed that, but like I said, I can't actually exercise these new lines

Reply:
Oh sorry, I was also testing this prior to the previous PR's change... let me see what my diff looked like...

Reply:
Okay, so the other way I was tested was doing the `mv config/database.yml ...` and then doing `bundle exec rake environment`.

Reply:
Was also testing with

```console
$ bundle exec rake test:verify_no_db_access_loading_rails_environment
```

Since the `.with_dummy_database_url_configuration` is used for stuff like building UI assets.

Reply:
I'll give those a shot...thanks

Reply:
`rake test:verify_no_db_access_loading_rails_environment` passes with this PR as is (without your addition).  `rake environment` fails with a missing config/database.yml, but not on the code in question (it fails much earlier).

I still can't exercise the code in question. For example, if instead I put

```
      rescue => err
        raise "XXX HI - #{err}"
```

I can't seem to find a way to get the XXX error.

Reply:
So talking with @Fryguy in DMs, I think this is pretty hard to reach in this class, so adding the extra `rescue` is probably not necessary or a use case we can currently hit.  That said, I also can't see it hurting anything either, since if code surrounding it would ever change to allow the lacking of a `config/database.yml` to be missing, then this wouldn't half execution (which seems like a valid use case for this class in particular).

That said, it isn't a hill I am willing to die on, so I am going to approve this PR, but leave this "unresolved" just so the comments for this are more prevalent for those looking back.